output/json/json.go

```diff
@@ -171,6 +170,19 @@ func (o *Output) handleMetric(m *metrics.Metric, jw *jwriter.Writer) {
 	}
 	o.seenMetrics[m.Name] = struct{}{}
 
-	wrapMetric(m).MarshalEasyJSON(jw)
+	wrapped := metricEnvelope{
+		Type:   "Metric",
+		Metric: m.Name,
+	}
+	wrapped.Data.Name = m.Name
+	wrapped.Data.Type = m.Type
+	wrapped.Data.Contains = m.Contains
+	wrapped.Data.Submetrics = m.Submetrics
```

Comment:
hmm I think this may be race-y as well :thinking: can you add a sub-metric to the test?

Reply:
It really shouldn't be as all the submetrics are *now* initialized early on in the starting of k6 as I mention in https://github.com/grafana/k6/pull/2596#discussion_r919808764 

https://github.com/grafana/k6/blob/7cdc0b18226a37f8cb90cacb57f81232d2298158/metrics/engine/engine.go#L73-L89

Reply:
fair enough :+1:

Reply:
was about to resolve this, but adding submetrics to the test still is a good idea though :sweat_smile: just to ensure we output whatever we expect (and we don't break that in a few versions)

Reply:
Added